addons/addon-base-raas/packages/base-raas-services/lib/study/study-service.js

```diff
@@ -262,7 +262,10 @@ class StudyService extends Service {
         const permissions = await this.studyPermissionService.getRequestorPermissions(requestContext);
         if (permissions) {
           // We can't give duplicate keys to the batch get, so ensure that allowedStudies is unique
-          const allowedStudies = _.uniq(permissions.adminAccess.concat(permissions.readonlyAccess));
+          const adminAccess = permissions.adminAccess || [];
+          const readonlyAccess = permissions.readonlyAccess || [];
+          const readwriteAccess = permissions.readwriteAccess || [];
+          const allowedStudies = _.uniq([...adminAccess, ...readonlyAccess, ...readwriteAccess]);
           if (allowedStudies.length) {
             const rawResult = await this._getter()
```

Comment:
Something we need to address in the future => batch get has a limit of 100 items.  We need to improve on this logic and do chunking to handle more than 100 studies.

https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BatchGetItem.html